### PR TITLE
tools: logger: Skip double error logs to stdout

### PR DESCRIPTION
**Pull request by ktrzcinx**
When out_fd points standard output then error message shouldn't be
duplicated - duplicated messages only make logger output more messy.

Signed-off-by: Karol Trzcinski <karolx.trzcinski@linux.intel.com>

### CODE REVIEW

File: tools/logger/convert.c [hunk 1]
```diff
@@ -68,7 +68,7 @@ static void log_err(FILE *out_fd, const char *fmt, ...)
 	if (buff) {
 		vsprintf(buff, fmt, args);
 		fprintf(stderr, "%s", buff);
```
Comment (xiulipan) [inline tools/logger/convert.c]:
```suggestion
		fprintf(stderr, "Error: %s", buff);
```
And remove the Error in the original message, will make it easy to keep align for error message.

File: tools/logger/convert.c [hunk 2]
```diff
@@ -472,7 +472,7 @@ static int fetch_entry(const struct convert_config *config,
 			}
 			if (ret != size)
 				log_err(config->out_fd,
-					"Partial read of %u bytes of %lu.\n",
+					"error: Partial read of %u bytes of %lu.\n",
 					ret, size);
```
Comment (xiulipan) [inline tools/logger/convert.c]: I remember other error message prefix is `Error`, could you check above comment to warp the Error prefix in the `log_err`?

Comment (ktrzcinx) [reply tools/logger/convert.c]: In this file there are both versions in usage, `Error` and `error`

File: tools/logger/convert.c (hunk 1 above)
Comment (ktrzcinx) [reply tools/logger/convert.c]: Yes, I also thought about that. I'm sitting to work

File: tools/logger/convert.c (hunk 2 above)
Comment (ktrzcinx) [reply tools/logger/convert.c]: in `logger.c` `error: ` version is used consistently, so I stick to this.

File: tools/logger/convert.c [hunk 3]
```diff
@@ -68,7 +68,7 @@ static void log_err(FILE *out_fd, const char *fmt, ...)
 	if (buff) {
 		vsprintf(buff, fmt, args);
 		fprintf(stderr, "%s", buff);
-		if (out_fd) {
+		if (out_fd && out_fd != stderr && out_fd != stdout) {
```
Comment (lgirdwood) [inline tools/logger/convert.c]: Could do with an inline comment here since we are testing three conditions.

Comment (ktrzcinx) [reply tools/logger/convert.c]: added

**Comment by aiChaoSONG** [comment]
> LGTM! thanks for the quick fix.
> @Bin-QA @aiChaoSONG could you help to check if this update will influence any sof-test case

@xiulipan we are good with this PR.

**Comment by lgirdwood** [comment]
@wwittbrx @lbetlej I'm good to merge but cant see the issue with internal CI unless it is not expecting a string change ?

**Comment by jajanusz** [comment]
https://travis-ci.org/github/thesofproject/sof/builds/665823502 shows build as OK, however it wasn't reported to GitHub

**Pull request merged**